gamevault/PipeServiceHandler.cs

```diff
@@ -0,0 +1,750 @@
+        /// Whether or not the handler is ready for commands. If set to false (the default) then all commands sent to the main instance will be queued until ready.
+        /// </summary>
+        public bool IsReadyForCommands
+        {
+            get => _isReadyForCommands;
+            set
+            {
+                if (_isReadyForCommands == value)
+                    return;
+
+                if (value)
+                {
+                    isReadyForCommandsTCS?.TrySetResult();
+                }
+                else
+                {
+                    // If we somehow have an existing tcs, cancel it before making a new one
+                    isReadyForCommandsTCS?.TrySetCanceled();
+                    isReadyForCommandsTCS = new TaskCompletionSource();
+                }
+
+                _isReadyForCommands = value;
+            }
+        }
+
+        private PipeServiceHandler()
+        {
+        }
+
+        /// <summary>
+        /// Starts the handler
+        /// </summary>
+        public static void StartInstance()
+        {
+            if (Instance != null)
+                return;
+
+            Instance = new PipeServiceHandler();
+
+            Instance.RegisterUriScheme();
+
+            try
+            {
+                Instance.StartNamedPipeServer();
+            }
+            catch (Exception ex)
+            {
+                System.Windows.MessageBox.Show(ex.Message, "An error while startin the internal pipe server", System.Windows.MessageBoxButton.OK, System.Windows.MessageBoxImage.Error);
+            }
+        }
+
+        /// <summary>
+        /// Registers the URI scheme which lets other applications communicate with the application
+        /// </summary>
+        private void RegisterUriScheme()
+        {
+            string? executablePath = null;
+
+            // winexe does not easily support getting our .exe file
+            //executablePath = System.Reflection.Assembly.GetEntryAssembly()?.Location;
+            // so get it from the process
+            executablePath = System.Diagnostics.Process.GetCurrentProcess().MainModule?.FileName;
+
+            if (string.IsNullOrEmpty(executablePath))
+                return;
+
+#if WINDOWS
+            var view = Microsoft.Win32.RegistryView.Registry32;
+            if (Environment.Is64BitOperatingSystem)
+                view = Microsoft.Win32.RegistryView.Registry64;
+
+            using var root = Microsoft.Win32.RegistryKey.OpenBaseKey(Microsoft.Win32.RegistryHive.CurrentUser, view);
+            using var classes = root.OpenSubKey(@"Software\Classes", true)!;
+
+            var openString = $"\"{executablePath}\" --uridata \"%1\"";
+            using var existing = classes.OpenSubKey(@$"{GAMEVAULT_URI_SCHEME}\shell\open\command");
+
+            if (existing != null && existing.GetValue("")?.ToString() == openString)
+                return;
+
+            using var newEntry = classes.CreateSubKey(GAMEVAULT_URI_SCHEME);
```

Comment:
This will only work in the non-Microsoft Store version. The UWP packages from the Store "virtualize" some things. Among them appdata and registry. So, if you create a registry key, it will be ignored because the package somehow does its own thing. I have had this problem in the past with the app autostart. But Microsoft lets you control these things via the StoreContext. 
I'll have to sit down again and see how this works for the Uri theme (Something like this [Link](https://learn.microsoft.com/en-us/windows/uwp/launch-resume/handle-uri-activation)). 
If you are interested in how the autostart solution looked like, you can have a look at "AutostartHelper.cs".

Reply:
Also we should probably do both to support Non-Store Versions as well.

Reply:
Looks like you just add the extension for the protocol to your package.appxmanifest, then add an `OnActivated` to App.xaml.cs that looks something like

```
   protected override void OnActivated(IActivatedEventArgs args)
  {
      if (args.Kind == ActivationKind.Protocol)
      {
         ProtocolActivatedEventArgs eventArgs = args as ProtocolActivatedEventArgs;
         _ = Task.Run(async () => PipeServiceHandler.HandleMessage(eventArgs.Uri.AbsoluteUri));
      }
   }
```

Unless of course it spins up a new instance to handle the OnActivated, in which case you'd need to see what's being sent in to the Program.cs for command line arguments and then pass the information from there. Codeium says this is not the case though:
![image](https://github.com/Phalcode/gamevault-app/assets/4464735/fa78b088-93c5-4d47-a3a1-c35d247b2e06)

I'm not gonna make those changes since I don't have access to the package.appxmanifest or any tooling to remotely test UWP applications.

Would recommend merging this through to a new branch and then working on it from there in your own environment with full UWP capabilities.